src/OfficeHours.ts

```diff
@@ -19,29 +20,28 @@ interface OfficeHoursData {
     nextId: number;
 }
 
-export default class OfficeHours {
-    private bot: Discord.Client;
+export default class OfficeHours extends CommandHandler {
     private data: OfficeHoursData;
     private sharedSettings: SharedSettings;
+    private bot: Discord.Client;
 
     private guild: Discord.Guild;
 
-    constructor(bot: Discord.Client, sharedSettings: SharedSettings, officeHoursData: string) {
+    constructor(sharedSettings: SharedSettings, officeHoursData: string) {
+        super();
         console.log("Requested OfficeHours extension..");
-        this.bot = bot;
 
         this.data = fileBackedObject(officeHoursData);
         console.log("Successfully question file.");
 
         this.sharedSettings = sharedSettings;
 
-        this.bot.on("ready", this.onBot.bind(this));
-        this.bot.on("message", this.onCommand.bind(this));
-		// this.bot.on("channelUpdate", this.onChannelUpdate.bind(this));
+        // this.bot.on("channelUpdate", this.onChannelUpdate.bind(this));
     }
 
-    onBot() {
-        const mainGuild = this.bot.guilds.get(this.sharedSettings.server);
+    onReady(bot: Discord.Client) {
+        this.bot = bot;
```

Comment:
Any cleaner way to do this? This seems a bit hacky.

Reply:
Its either gonna end up in onReady or the constructor, and I'd prefer having the constructors as clean as possible :man_shrugging: 

(Tho I managed to remove it altogether from officehours, it still remains in some of the other classes)